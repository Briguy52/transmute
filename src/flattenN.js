// @flow
import curry from "./curry";
import type { Iterable } from "immutable";

/**
 * Flattens an iterable `depth` levels.
 *
 * @example
 * // return List [ 1, List [ 2, 3 ], 4, 5, 6 ]
 * flattenN(1, List.of(List.of(1, List.of(2, 3)), List.of(4, 5, 6)));
 */
function flattenN(depth: number, subject: Iterable<*, *>) {
  if (typeof depth !== "number" || depth <= 0) {
    throw new Error(
      `expected \`depth\` to be a number greater than 0 but got \`${depth}\``
    );
  }
  return subject.flatten(depth);
}

export default curry(flattenN);
