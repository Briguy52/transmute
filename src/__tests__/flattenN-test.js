import flattenN from "../flattenN";
import { List, Seq } from "immutable";

describe("transmute/flattenN", () => {
  it("flattens Lists to depth", () => {
    expect(
      flattenN(1, List.of(List.of(1, List.of(2, 3)), List.of(4, 5, 6)))
    ).toMatchSnapshot();
  });

  it("flattens Seqs to depth", () => {
    expect(
      flattenN(1, Seq.of(Seq.of(1, Seq.of(2, 3)), Seq.of(4, 5, 6)))
    ).toMatchSnapshot();
  });
});
