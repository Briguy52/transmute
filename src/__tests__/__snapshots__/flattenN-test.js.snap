// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/flattenN flattens Lists to depth 1`] = `
Immutable.List [
  1,
  Immutable.List [
    2,
    3,
  ],
  4,
  5,
  6,
]
`;

exports[`transmute/flattenN flattens Seqs to depth 1`] = `
Array [
  1,
  Array [
    2,
    3,
  ],
  4,
  5,
  6,
]
`;
